go-controller/pkg/ovn/master.go

```diff
@@ -772,6 +772,26 @@ func (oc *Controller) deleteNode(nodeName string, nodeSubnet, joinSubnet *net.IP
 		return fmt.Errorf("Failed to clean up node %s gateway: (%v)", nodeName, err)
 	}
 
+	chassisName, stderr, err := util.RunOVNSbctl("--data=bare", "--no-heading",
+		"--columns=name", "find", "Chassis",
+		"hostname="+nodeName)
+	if err != nil {
+		klog.Errorf("Failed to get chassis name for logical switch %s: stderr: %q, error: %v",
+			nodeName, stderr, err)
+		return err
+	}
+
+	if chassisName == "" {
+		klog.Warningf("Chassis name is empty for logical switch: %s", nodeName)
+	} else {
+		_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
+		if err != nil {
+			klog.Errorf("Failed to delete chassis with name %s for logical switch %s: stderr: %q, error: %v",
+				chassisName, nodeName, stderr, err)
+			return err
+		}
+	}
+
 	return nil
```

Comment:
to be consistent with other deletion we do in that  function, please move L775-794 into deleteNodeChassis() (or some such)

Reply:
ack